agent/drive_handler.go

```diff
@@ -229,6 +218,45 @@ func (dh driveHandler) MountDrive(ctx context.Context, req *drivemount.MountDriv
 		drive.Path(), req.DestinationPath)
 }
 
+func (dh driveHandler) UnmountDrive(ctx context.Context, req *drivemount.UnmountDriveRequest) (*empty.Empty, error) {
+	drive, ok := dh.GetDrive(req.DriveID)
+	if !ok {
+		return nil, fmt.Errorf("drive %q could not be found", req.DriveID)
+	}
+
+	// Do a basic check that we won't be mounting over any important system directories
+	if err := checkImportantDirs(req.DestinationPath); err != nil {
+		return nil, err
+	}
```

Comment:
Does it need to check the path here?

`UnmoutDriveRequest { DriveID = (some valid ID), DestinationPath = (some invalid path) }` will be rejected by this check, but it wouldn't harm anything since the valid ID is anyway pointing the valid (non-important) directory.

Reply:
+1 I'm not sure there's a need to check this here too as the mount has already occurred.